src/types/fee_multiple.ts

```diff
@@ -0,0 +1,42 @@
+import BigNumber from 'bignumber.js';
+
+export class FeeMultiple {
+	constructor(private readonly feeMultiple: number) {
+		if (this.feeMultiple < 1.0 || Number.isNaN(feeMultiple) || !Number.isFinite(feeMultiple)) {
```

Comment:
I think we don't need to check for the maximum. There's no maximum boost, or there is it?

Reply:
@matibat: the theorical max is MAX_SAFE_INTEGER for now
```ts
!Number.isFinite(feeMultiple)
```
**Edit** See the test case of "can boost large rewards"

Reply:
_and the wallet balance_ :rofl: